include/seqan3/io/alignment_file/format_bam.hpp

```diff
@@ -0,0 +1,1061 @@
+        tag += static_cast<uint16_t>(*std::ranges::begin(stream_view));
+        std::ranges::next(std::ranges::begin(stream_view)); // skip char read before
+        char type_id = static_cast<char>(*std::ranges::begin(stream_view));
+        std::ranges::next(std::ranges::begin(stream_view)); // skip char read before
+
+        switch (type_id)
+        {
+            case 'A' : // char
+            {
+                target[tag] = static_cast<char>(*std::ranges::begin(stream_view));
+                std::ranges::next(std::ranges::begin(stream_view)); // skip char that has been read
+                break;
+            }
+            // all integer sizes are possible
+            case 'c' : // int8_t
+            {
+                int8_t tmp;
+                read_field(stream_view, tmp);
+                target[tag] = static_cast<int32_t>(tmp); // readable sam format only allows int32_t
+                break;
+            }
+            case 'C' : // uint8_t
+            {
+                uint8_t tmp;
+                read_field(stream_view, tmp);
+                target[tag] = static_cast<int32_t>(tmp); // readable sam format only allows int32_t
+                break;
+            }
+            case 's' : // int16_t
+            {
+                int16_t tmp;
+                read_field(stream_view, tmp);
+                target[tag] = static_cast<int32_t>(tmp); // readable sam format only allows int32_t
+                break;
+            }
+            case 'S' : // uint16_t
+            {
+                uint16_t tmp;
+                read_field(stream_view, tmp);
+                target[tag] = static_cast<int32_t>(tmp); // readable sam format only allows int32_t
+                break;
+            }
+            case 'i' : // int32_t
+            {
+                int32_t tmp;
+                read_field(stream_view, tmp);
+                target[tag] = std::move(tmp); // readable sam format only allows int32_t
+                break;
+            }
+            case 'I' : // uint32_t
+            {
+                uint32_t tmp;
+                read_field(stream_view, tmp);
+                target[tag] = static_cast<int32_t>(tmp); // readable sam format only allows int32_t
+                break;
+            }
+            case 'f' : // float
+            {
+                float tmp;
+                read_field(stream_view, tmp);
+                target[tag] = tmp;
+                break;
+            }
+            case 'Z' : // string
+            {
+                std::string tmp;
+                while (!is_char<'\0'>(*std::ranges::begin(stream_view)))
+                {
+                    tmp.push_back(*std::ranges::begin(stream_view));
+                    std::ranges::next(std::ranges::begin(stream_view));
+                }
+                std::ranges::next(std::ranges::begin(stream_view)); // skip \0
+                target[tag] = std::move(tmp);
```

Comment:
I think it might be faster to use a buffer here and copy out instead of creating a new string and moving that out. Because in the current situation it might do repeated allocations when resizing the string, whereas when you have a buffer it will have a large size after a while and you will always only have exactly one alloc (when copying).